File: Drivo.WebAPI/Services/DrivingsService.cs
using Drivo.Entities;
using Drivo.Responses;
using Drivo.WebAPI.Repositories;

namespace Drivo.WebAPI.Services;

public class DrivingsService
{
    public DrivingsService(DrivingsRepository drivingsRepository)
    {
        DrivingsRepository = drivingsRepository;
    }

    private DrivingsRepository DrivingsRepository { get; }

    public async Task<DrivingEntity> GetDrivingByIdAsync(int drivingId)
    {
        return await DrivingsRepository.GetDrivingByIdAsync(drivingId);
    }

    public async Task<ActionResponse> AddDrivingAsync(DrivingEntity driving)
    {
        return await DrivingsRepository.AddDrivingAsync(driving);
    }

    public async Task<ActionResponse> UpdateDrivingAsync(DrivingEntity driving)
    {
        return await DrivingsRepository.UpdateDrivingAsync(driving);
    }

    public async Task<ActionResponse> RemoveDrivingAsync(int drivingId)
    {
        return await DrivingsRepository.RemoveDrivingAsync(await GetDrivingByIdAsync(drivingId));
    }
}

File: Drivo/src/app/services/drivings.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { firstValueFrom } from 'rxjs';
import { DrivingEntity } from 'src/entities/DrivingEntity';
import { environment } from 'src/environments/environment';
import { ActionResponse } from 'src/responses/action.response';

@Injectable({
  providedIn: 'root'
})
export class DrivingService {
  
  constructor(private httpClient: HttpClient) {}

  public async getDrivingsAsync(): Promise<Array<DrivingEntity>> {
    return await firstValueFrom(this.httpClient.get<Array<DrivingEntity>>(`${environment.apiUri}/Drivings`));
  }

  public async addDrivingAsync(driving: DrivingEntity): Promise<ActionResponse> {
    return await firstValueFrom(this.httpClient.post<ActionResponse>(`${environment.apiUri}/Drivings`, driving));
  }

  public async updateDrivingAsync(driving: DrivingEntity): Promise<ActionResponse> {
    return await firstValueFrom(this.httpClient.put<ActionResponse>(`${environment.apiUri}/Drivings`, driving));
  }

  public async removeDrivingAsync(drivingId: number): Promise<ActionResponse> {
    return await firstValueFrom(this.httpClient.delete<ActionResponse>(`${environment.apiUri}/Drivings/${drivingId}`));
  }

}
